diff --git a/sound.js b/sound.js
index c9bcbaf1e397a7a9600a08912c0f03f73b82678f..8b3f85f4e466fa243202f207aa661472946ad8a9 100644
--- a/sound.js
+++ b/sound.js
@@ -4,7 +4,8 @@ var ReactNative = require('react-native');
 var RNSound = ReactNative.NativeModules.RNSound;
 var IsAndroid = RNSound.IsAndroid;
 var IsWindows = RNSound.IsWindows;
-var resolveAssetSource = require("react-native/Libraries/Image/resolveAssetSource");
+var resolveAssetSourceModule = require("react-native/Libraries/Image/resolveAssetSource");
+var resolveAssetSource = resolveAssetSourceModule.default || resolveAssetSourceModule;
 var eventEmitter = new ReactNative.NativeEventEmitter(RNSound);
 
 var nextKey = 0;
